module/VuFindApi/src/VuFindApi/Controller/AdminApiController.php

```diff
@@ -0,0 +1,173 @@
+     *
+     * @return \Laminas\Http\Response
+     */
+    public function clearCacheAction()
+    {
+        $this->disableSessionWrites();
+        $this->determineOutputMode();
+
+        if ($result = $this->isAccessDenied($this->cacheAccessPermission)) {
+            return $result;
+        }
+
+        try {
+            $cacheList = $this->getRequest()->getQuery()->get('id')
+                ?: $this->getDefaultCachesToClear();
+            foreach ((array)$cacheList as $id) {
+                $cache = $this->cacheManager->getCache($id);
+                if (!($cache instanceof \Laminas\Cache\Storage\FlushableInterface)) {
+                    throw new \Exception("Cache $id is not flushable");
+                }
+                $cache->flush();
+            }
+        } catch (\Exception $e) {
+            return $this->output([], self::STATUS_ERROR, 500, $e->getMessage());
+        }
+
+        return $this->output([], self::STATUS_OK);
+    }
+
+    /**
+     * Get Swagger specification JSON fragment for services provided by the
+     * controller
+     *
+     * @return string
+     */
+    public function getSwaggerSpecFragment()
+    {
+        $spec = [];
+        if (!$this->isAccessDenied($this->cacheAccessPermission)) {
+            $defaultCaches = implode(',', $this->getDefaultCachesToClear());
+            $spec['paths']['/admin/cache']['delete'] = [
+                'summary' => 'Clear caches',
+                'description' => 'Flushes the specified caches',
+                'parameters' => [
+                    [
+                        'name' => 'id[]',
+                        'in' => 'query',
+                        'description' => 'Caches to clear. By default the following'
+                            . " caches are cleared: $defaultCaches",
+                        'required' => false,
+                        'type' => 'array',
+                        'collectionFormat' => 'multi',
+                        'items' => [
+                            'type' => 'string'
+                        ]
+                    ]
+                ],
+                'tags' => ['admin'],
+                'responses' => [
+                    '200' => [
+                        'description' => 'An OK response',
+                        'schema' => [
+                            '$ref' => '#/definitions/Success'
+                        ]
+                    ],
+                    'default' => [
+                        'description' => 'Error',
+                        'schema' => [
+                            '$ref' => '#/definitions/Error'
+                        ]
+                    ]
+                ]
+            ];
+        }
+
+        // Admin API endpoints are not published
+        return json_encode($spec);
+    }
+
+    /**
+     * Get an array of caches to clear by default
+     *
+     * @return array
+     */
+    protected function getDefaultCachesToClear(): array
+    {
+        return array_diff(
```

Comment:
Could we filter the unflushable caches at this layer to reduce the odds of errors?

Reply:
Good idea, done.